integrations/mill-bloop/src/main/scala/bloop/integrations/mill/MillBloop.scala

```diff
@@ -0,0 +1,138 @@
+package bloop.integrations.mill
+
+import _root_.mill._
+import _root_.mill.define._
+import _root_.mill.scalalib._
+import _root_.mill.eval.Evaluator
+
+import ammonite.ops._
+
+object Bloop extends ExternalModule {
+
+  def install(ev: Evaluator[Any]) = T.command {
+    val bloopDir = pwd / ".bloop"
+    mkdir(bloopDir)
+
+    val rootModule = ev.rootModule
+    val modules = rootModule.millInternal.segmentsToModules.values.collect {
+      case m: scalalib.JavaModule => m
+    }.toSeq
+
+    Task.traverse(modules)(genBloopConfig(bloopDir, _))
+  }
+
+  implicit def millScoptEvaluatorReads[T] = new mill.main.EvaluatorScopt[T]()
+
+  def genBloopConfig(bloopDir: Path, module: JavaModule): Task[Path] = {
+
+    import bloop.config.Config
+
+    def name(m: JavaModule) = m.millModuleSegments.render
+    def out(m: JavaModule) = bloopDir / "out" / m.millModuleSegments.render
+    def analysisOut(m: JavaModule) = out(m) / Config.Project.analysisFileName(name(m))
+    def classes(m: JavaModule) = out(m) / "classes"
+
+    val javaConfig = module.javacOptions.map(opts => Config.Java(options = opts.toArray))
+
+    val scalaConfig = module match {
+      case s: ScalaModule =>
+        T.task {
+          val pluginOptions = s.scalacPluginClasspath().map { pathRef =>
+            s"-Xplugin:${pathRef.path}"
+          }
+
+          Config.Scala(
+            organization = "org.scala-lang",
+            name = "scala-compiler",
+            version = s.scalaVersion(),
+            options = (s.scalacOptions() ++ pluginOptions).toArray,
+            jars = s.scalaCompilerClasspath().map(_.path.toNIO).toArray
+          )
+        }
+      case _ => T.task(Config.Scala.empty)
+    }
+
+    val jvmConfig = T.task {
+      Config.Jvm(
+        home = T.ctx().env.get("JAVA_HOME").map(s => Path(s).toNIO),
+        options = module.forkArgs().toArray
+      )
+    }
+
+    val testConfig = module match {
+      case m: TestModule =>
+        T.task {
+          Config.Test(
+            frameworks = m
+              .testFrameworks()
+              .map(f => Config.TestFramework(List(f)))
+              .toArray,
+            options = Config.TestOptions(
+              excludes = List(),
+              arguments = List()
+            )
+          )
+        }
+      case _ => T.task(Config.Test.empty)
+    }
+
+    val ivyDepsClasspath =
+      module
+        .resolveDeps(T.task { module.compileIvyDeps() ++ module.transitiveIvyDeps() })
+        .map(_.map(_.path).toSeq)
+
+    def transitiveClasspath(m: JavaModule): Task[Seq[Path]] = T.task {
```

Comment:
@jvican this is what I had to re-implement (rather than calling the mill one directly) to make sure the `install` task does not call upon the `compile` task

Reply:
Great! I cannot get back to the rebase tonight but wil do tomorrow morning.